docs/proposals/all-in-one-volume.md

```diff
@@ -0,0 +1,150 @@
+## Abstract
+
+A proposal for enabling users to create a volume populated from secrets,
+configmaps, and downward API.
+
+## Motivation
+
+This new feature will allow the user a method of combining data from different
+sources within the same volume, which is not something that can be done
+currently.
+
+## Constraints and Assumptions
+
+There can not be an overlap of resources with the same paths. If a conflict
```

Comment:
1.  Existing volume types must remain unchanged for backward compatibility purposes
2.  There should be a new volume type for this functionality; no other API changes